software/18.Kafka/README.md

```diff
@@ -0,0 +1,214 @@
+- Since this restaurant has a lot of orders, you'll help them scale up using Kafka
+- Once the order has been published into Kafka, two consumers will consume this order
+- The cook will prepare the order
+- The manager keeps track of the orders and record it in a file (like a database)
+
+> But what *is* Kafka ? What do I need this anyway ?
+
+Kafka is a [message broker](https://www.ibm.com/cloud/learn/message-brokers), meaning that someone is responsible for sending messages to Kafka - the producer - and someone else will receive these messages - the consumer.
+Other message brokers exists, like [RabbitMQ](https://www.rabbitmq.com/), but Kafka is much more resilient, because it is designed to support stream interruptions.
+
+I'm not going to lie, Kafka is a bit overkill for this workshop.
+Well it is still a good way to learn its behaviour and how to interact with Kafka using your own producer / consumer.
+
+At the end of the workshop, you'll be able to scale up even further this application with more producers / consumers.
+
+## Step 0 - Setup
+
+Please follow the instructions available [here](./SETUP.md).
+
+## Step 1 - Start Kafka and ZooKeeper
+
+### :bookmark_tabs: **Description**:
+You'll need to start Kafka in a docker container, but Kafka actually relies on ZooKeeper, which you can learn more about [here](http://cloudurable.com/blog/kafka-architecture/index.html).
+
+Those services will start into docker containers.
+
+### :pushpin: **Tasks**:
+- Create a `docker-compose.yaml` file
+
+In this file, create a `zookeeper` service:
+- It uses the following docker image: `confluentinc/cp-zookeeper:latest`
+- It binds the container port `22181` to your local `22181` port
+- Initialize the required environment variables you can find [here](https://docs.confluent.io/platform/current/installation/docker/config-reference.html#zk-configuration)
+
+Then, create a second service, named `kafka`:
+- It uses the following docker image: `confluentinc/cp-kafka:latest`
+- It binds the following container ports to your local ports:
+  - 29092 to 29092, the internal listener
+  - 9092 to 9092, the external listener
+- Initialize the required environment variables you can find [here](https://docs.confluent.io/platform/current/installation/docker/config-reference.html#confluent-ak-configuration)
+- It depends on the `zookeeper` service
+
+### :books: **Documentation**:
+
+- [Control startup and shutdown order in docker compose](https://docs.docker.com/compose/startup-order/)
+- [Kafka listeners explained](https://www.confluent.io/blog/kafka-listeners-explained/)
+- [Kafka configuration properties](https://jaceklaskowski.gitbooks.io/apache-kafka/content/kafka-properties.html)
+
+### :heavy_check_mark: **Validation**:
+
+Run `docker-compose up --build`, wait 30 seconds and in a new terminal run `docker-compose logs kafka | grep -i 
```

Comment:
I think it will be simpler to use the `-d` argument, this way, you will certainly avoid the problem of the guy who stops his containers with a `^C` or closes his terminal :rofl: